addons/l10n_ch/models/account_invoice.py

```diff
@@ -71,6 +71,19 @@ def _compute_l10n_ch_isr_number(self):
         this number with zeros. The last character of the ISR number is the result
         of a recursive modulo 10 on its first 26 characters.
         """
+        for record in self:
+            if record.name and record.invoice_partner_bank_id and record.invoice_partner_bank_id.l10n_ch_postal:
+                invoice_issuer_ref = record.invoice_partner_bank_id.l10n_ch_postal.ljust(l10n_ch_ISR_NUMBER_ISSUER_LENGTH, '0')
+                invoice_ref = re.sub('[^\d]', '', record.name)
+                #We only keep the last digits of the sequence number if it is too long
+                invoice_ref = invoice_ref[-l10n_ch_ISR_NUMBER_ISSUER_LENGTH:]
+                internal_ref = invoice_ref.zfill(l10n_ch_ISR_NUMBER_LENGTH - l10n_ch_ISR_NUMBER_ISSUER_LENGTH - 1) # -1 for mod10r check character
+                record.l10n_ch_isr_number = mod10r(invoice_issuer_ref + internal_ref)
+            else:
+                record.l10n_ch_isr_number = False
+
+    @api.depends('name', 'invoice_partner_bank_id.l10n_ch_postal')
+    def _compute_l10n_ch_isr_number_spaced(self):
```

Comment:
@jva-odoo Is not it better to depend on l10n_ch_isr_number here in the api.depends?

Reply:
Yeah right

Reply:
@jco-odoo Done 👍

Reply:
@jva-odoo What did you change exactly?

Reply:
@jva-odoo Did you test the changes?

Reply:
@jco-odoo put l10n_ch_isr_number in depends

Reply:
It says  @api.depends('name', 'invoice_partner_bank_id.l10n_ch_postal')

Reply:
@jco-odoo  My change is removed by your force-pushed

Reply:
@jva-odoo Damn, ok. Can you add it again?

Reply:
@jco-odoo ok 👍